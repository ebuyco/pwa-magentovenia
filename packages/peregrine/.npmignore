.eslintrc.js
.storybook
jest.config.js
/{dist,src}/**/__{docs,helpers,mocks,stories,tests}__/**
/scripts
