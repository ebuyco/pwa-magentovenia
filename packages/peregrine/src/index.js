import * as RestApi from './RestApi';
import * as Util from './util';

export { default as ContainerChild } from './ContainerChild';
export { default as List, Items, Item } from './List';
export { default as Page } from './Page';
export { default as Price } from './Price';
export { default as Router } from './Router';
export { RestApi };
export { Util };
