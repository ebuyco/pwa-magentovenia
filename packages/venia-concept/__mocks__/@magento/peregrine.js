import React from 'react';
import { BrowserPersistence } from './util/simplePersistence';

const peregrine = jest.requireActual('../../../peregrine/src');

const mockRequest = jest.fn();

const RestApi = {
    Magento2: {
        request: mockRequest
    }
};

const Util = {
    BrowserPersistence: BrowserPersistence
};

/**
 * the Price component from @magento/peregrine
 * has browser-specific functionality and cannot
 * currently by rendered in the test environment
 */
const Price = () => <div />;

module.exports = {
    ...peregrine,
    mockRequest,
    RestApi,
    Util,
    Price
};
