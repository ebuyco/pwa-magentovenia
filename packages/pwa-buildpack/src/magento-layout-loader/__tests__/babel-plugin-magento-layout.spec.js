const dedent = require('dedent');
const babel = require('@babel/core');
const jsxTransform = require('@babel/plugin-transform-react-jsx');
const plugin = require('../babel-plugin-magento-layout');

const transform = (opts, input, pragma) => {
    const { code } = babel.transform(input, {
        plugins: [[jsxTransform, { pragma }], [plugin, opts]]
    });
    return code;
};

test('Warns if "data-mid" is found on a Composite Component', () => {
    const onWarning = jest.fn();
    const opts = {
        config: {
            'product.page': [
                {
                    operation: 'removeContainer',
                    targetContainer: 'product.page'
                }
            ]
        },
        onWarning
    };
    transform(
        opts,
        dedent`
            import React from 'react';
            <Foo data-mid='bizz.buzz' />
        `
    );
    expect(onWarning).toHaveBeenCalledWith(
        expect.stringContaining('"data-mid" found on a Composite Component')
    );
});

test('onWarning callback invoked when data-mid is a dynamic value', () => {
    const onWarning = jest.fn();
    const opts = {
        config: {
            'product.page.pricing': []
        },
        onWarning
    };
    transform(
        opts,
        dedent`
            import React from 'react';
            <div data-mid={someRef} />
        `
    );
    expect(onWarning).toHaveBeenCalledWith(
        expect.stringContaining('Expected "data-mid" to be a literal string')
    );
});

test('Operations work when createElement is used instead of React.createElement', () => {
    const opts = {
        config: {
            'product.page': [
                {
                    operation: 'removeContainer',
                    targetContainer: 'product.page'
                }
            ]
        }
    };
    const result = transform(
        opts,
        dedent`
            import { createElement } from 'react';
            <div>
                Should remove container below
                <div data-mid='product.page' />
            </div>
        `,
        'createElement'
    );
    expect(result).toMatchSnapshot();
});

test('"removeContainer" operation removes container', () => {
    const opts = {
        config: {
            'product.page': [
                {
                    operation: 'removeContainer',
                    targetContainer: 'product.page'
                }
            ]
        }
    };
    const result = transform(
        opts,
        dedent`
            import React from 'react';
            <div>
                Should remove container below
                <div data-mid='product.page' />
            </div>
    `
    );
    expect(result).toMatchSnapshot();
});

test('"removeChild" operation removes specified child, but not other children', () => {
    const opts = {
        config: {
            'product.page.pricing': [
                {
                    operation: 'removeChild',
                    targetContainer: 'product.page.pricing',
                    targetChild: 'product.page.pricing.add.button'
                }
            ]
        }
    };
    const result = transform(
        opts,
        dedent`
            import React from 'react';
            import { ContainerChild } from '@magento/peregrine';
            <div data-mid='product.page.pricing'>
                <ContainerChild
                    id='product.page.pricing.add.button'
                    render={() =>
                        <div>I should not render</div>
                    }
                />
                <ContainerChild
                    id='product.page.pricing.some.other.button'
                    render={() => <div>I should render</div>}
                />
            </div>
        `
    );
    expect(result).toMatchSnapshot();
});

test('A removeChild operation works when ContainerChild is aliased in import', () => {
    const opts = {
        config: {
            'product.page.pricing': [
                {
                    operation: 'removeChild',
                    targetContainer: 'product.page.pricing',
                    targetChild: 'product.page.pricing.add.button'
                }
            ]
        }
    };
    const result = transform(
        opts,
        dedent`
            import React from 'react';
            import { ContainerChild as CC } from '@magento/peregrine';
            <div data-mid='product.page.pricing'>
                <CC
                    id='product.page.pricing.add.button'
                    render={() =>
                        <div>I should not render</div>
                    }
                />
            </div>
        `
    );
    expect(result).toMatchSnapshot();
});

test('"insertBefore" injects an extension before its target ContainerChild', () => {
    const opts = {
        config: {
            'product.page.pricing': [
                {
                    operation: 'insertBefore',
                    targetContainer: 'product.page.pricing',
                    targetChild: 'product.page.pricing.add.button',
                    componentPath: '/Users/person/components/SomeComponent.js'
                }
            ]
        }
    };
    const result = transform(
        opts,
        dedent`
            import React from 'react';
            import { ContainerChild } from '@magento/peregrine';
            <div data-mid='product.page.pricing'>
                <ContainerChild
                    id='product.page.pricing.add.button'
                    render={() => {}}
                />
            </div>
        `
    );
    expect(result).toMatchSnapshot();
});

test('"insertAfter" injects an extension after its target ContainerChild', () => {
    const opts = {
        config: {
            'product.page.pricing': [
                {
                    operation: 'insertAfter',
                    targetContainer: 'product.page.pricing',
                    targetChild: 'product.page.pricing.add.button',
                    componentPath: '/Users/person/components/SomeComponent.js'
                }
            ]
        }
    };
    const result = transform(
        opts,
        dedent`
            import React from 'react';
            import { ContainerChild } from '@magento/peregrine';
            <div data-mid='product.page.pricing'>
                <ContainerChild
                    id='product.page.pricing.add.button'
                    render={() => {}}
                />
            </div>
        `
    );
    expect(result).toMatchSnapshot();
});
