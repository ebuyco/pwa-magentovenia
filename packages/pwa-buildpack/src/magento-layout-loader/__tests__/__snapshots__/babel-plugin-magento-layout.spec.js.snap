// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"insertAfter" injects an extension after its target ContainerChild 1`] = `
"\\"use strict\\";

var _SomeComponent = _interopRequireDefault(require(\\"/Users/person/components/SomeComponent.js\\"));

var _react = _interopRequireDefault(require(\\"react\\"));

var _peregrine = require(\\"@magento/peregrine\\");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

_react.default.createElement(\\"div\\", {
  \\"data-mid\\": \\"product.page.pricing\\"
}, _react.default.createElement(_peregrine.ContainerChild, {
  id: \\"product.page.pricing.add.button\\",
  render: () => {}
}), _react.default.createElement(_SomeComponent.default, null));"
`;

exports[`"insertBefore" injects an extension before its target ContainerChild 1`] = `
"\\"use strict\\";

var _SomeComponent = _interopRequireDefault(require(\\"/Users/person/components/SomeComponent.js\\"));

var _react = _interopRequireDefault(require(\\"react\\"));

var _peregrine = require(\\"@magento/peregrine\\");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

_react.default.createElement(\\"div\\", {
  \\"data-mid\\": \\"product.page.pricing\\"
}, _react.default.createElement(_SomeComponent.default, null), _react.default.createElement(_peregrine.ContainerChild, {
  id: \\"product.page.pricing.add.button\\",
  render: () => {}
}));"
`;

exports[`"removeChild" operation removes specified child, but not other children 1`] = `
"\\"use strict\\";

var _react = _interopRequireDefault(require(\\"react\\"));

var _peregrine = require(\\"@magento/peregrine\\");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

_react.default.createElement(\\"div\\", {
  \\"data-mid\\": \\"product.page.pricing\\"
}, _react.default.createElement(_peregrine.ContainerChild, {
  id: \\"product.page.pricing.some.other.button\\",
  render: () => _react.default.createElement(\\"div\\", null, \\"I should render\\")
}));"
`;

exports[`"removeContainer" operation removes container 1`] = `
"\\"use strict\\";

var _react = _interopRequireDefault(require(\\"react\\"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

_react.default.createElement(\\"div\\", null, \\"Should remove container below\\");"
`;

exports[`A removeChild operation works when ContainerChild is aliased in import 1`] = `
"\\"use strict\\";

var _react = _interopRequireDefault(require(\\"react\\"));

var _peregrine = require(\\"@magento/peregrine\\");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

_react.default.createElement(\\"div\\", {
  \\"data-mid\\": \\"product.page.pricing\\"
});"
`;

exports[`Operations work when createElement is used instead of React.createElement 1`] = `
"\\"use strict\\";

var _react = require(\\"react\\");

(0, _react.createElement)(\\"div\\", null, \\"Should remove container below\\");"
`;
